js/main.js

```diff
@@ -19,58 +20,84 @@ function handleNewPhotoURL(event) {
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.editing.entryID === data.entries[i].entryID) {
+        var indexOfEntry = i;
+        data.entries[i].title = $newEntryForm.elements.title.value;
+        data.entries[i].image = $newEntryForm.elements.image.value;
+        data.entries[i].notes = $newEntryForm.elements.notes.value;
+      }
+    }
+    var $entry = document.querySelector('#id' + data.editing.entryID);
+    $entry.replaceWith(generateEntryDOM(data.entries[indexOfEntry]));
+    data.editing = null;
+  } else {
+    var newEntry = {
+      title: $newEntryForm.elements.title.value,
+      image: $newEntryForm.elements.image.value,
+      notes: $newEntryForm.elements.notes.value,
+      entryID: data.nextEntryId
+    };
+    data.nextEntryId++;
+    data.entries.unshift(newEntry);
+    $entryList.prepend(generateEntryDOM(newEntry));
+  }
   var $newPhoto = document.querySelector('.new-entry-img');
   $newPhoto.setAttribute('src', 'images/placeholder-image-square.jpg');
   $newEntryForm.reset();
-  $entryList.prepend(generateEntryDOM(newEntry));
   switchView('entries');
+  var $displayMessage = document.querySelector('.no-entries-msg');
+  if ($displayMessage) $displayMessage.remove();
 }
 
 function generateEntryDOM(entry) {
 // <li>
 //   <div class="row">
 //     <div class="column-half img-container">
-//       <img src="">
+//       <img class="entry-image" src="">
 //     </div>
-//     <div class="column-half">
-//       <h2></h2>
-//       <p></p>
+//     <div class="column-half"> relative position
+//       <h2 class="entry-title"></h2>
+//       <p class="entry-notes"></p>
+//       <img> (edit icon; absolute position)
 //     </div>
 //   </div>
 // </li>
 
   var $newImage = document.createElement('img');
   $newImage.setAttribute('src', entry.image);
+  $newImage.className = 'entry-image';
 
   var $imageBox = document.createElement('div');
   $imageBox.className = 'column-half img-container';
   $imageBox.appendChild($newImage);
 
   var $newTitle = document.createElement('h2');
   $newTitle.textContent = entry.title;
+  $newTitle.className = 'entry-title';
 
   var $newDescription = document.createElement('p');
   $newDescription.innerText = entry.notes;
+  $newDescription.className = 'entry-notes';
+
+  var $editIcon = document.createElement('img');
+  $editIcon.setAttribute('src', 'images/edit-icon.png');
+  $editIcon.className = 'edit-icon';
 
   var $textBox = document.createElement('div');
-  $textBox.className = 'column-half';
+  $textBox.className = 'column-half text-box';
   $textBox.appendChild($newTitle);
   $textBox.appendChild($newDescription);
+  $textBox.appendChild($editIcon);
 
   var $listContents = document.createElement('div');
   $listContents.className = 'row';
   $listContents.appendChild($imageBox);
   $listContents.appendChild($textBox);
 
   var $newListEntry = document.createElement('li');
+  $newListEntry.setAttribute('id', 'id' + entry.entryID.toString());
```

Comment:
Instead of using the `id` attribute (which is meant to be used to label individual DOM elements), you should add a new data attribute (maybe `data-entry-id`) that you can just store the id in directly.